src/alto-ui/Form/DatePicker/DatePicker.js

```diff
@@ -176,25 +185,21 @@ function DatePicker(props) {
         <DatePickerHeader
           date={date}
           id={id}
-          onChange={values => setDate(d => d.set(values))}
-          onChangeTime={values => {
-            const newDate = date.set(values);
-            setDate(newDate);
-            handleChange(newDate.toJSDate());
-          }}
+          onChange={onChangeDatetimeHeader}
+          onChangeTime={onChangeDatetimeHeader}
           datetime={datetime}
         />
         <div className="DatePicker__days">
           <DayPicker
             month={date.toJSDate()}
             canChangeMonth={false}
             captionElement={() => null}
-            onDayClick={d => {
-              const newDate = DateTime.fromJSDate(d);
+            onDayClick={jsDate => {
               if (datetime) {
-                newDate.set({ hour: date.hour, minute: date.minute });
+                const { hour, minute } = date;
+                jsDate.setHours(hour);
```

Comment:
create new date - do not mutate parameter.

Reply:
done